extensions/1password/src/v8/utils.ts

```diff
@@ -58,30 +58,56 @@ export function op(args: string[]) {
   throw Error("1Password CLI is not found!");
 }
 
+const CACHE_TIMEOUTS: { [key: string]: number } = {
+  [CATEGORIES_CACHE_NAME]: 1000 * 60 * 10,
+  [ITEMS_CACHE_NAME]: 1000 * 60 * 10,
+  [ACCOUNT_CACHE_NAME]: 1000 * 60 * 10,
+};
+
+const DEFAULT_CACHE_TIMEOUT = 1000 * 60 * 10;
+
 export function useOp<T>(args: string[], cacheKey?: string) {
   const [data, setData] = useState<T>();
   const [error, setError] = useState<unknown>();
   const [isLoading, setIsLoading] = useState<boolean>(true);
 
   useEffect(() => {
-    if (cacheKey && cache.has(cacheKey)) {
-      setIsLoading(false);
-      return setData(JSON.parse(cache.get(cacheKey) as string));
-    }
-
-    try {
-      const items = op([...args, "--format=json"]);
+    const fetchData = async () => {
+      setIsLoading(true);
+      try {
+        const items = op([...args, "--format=json"]);
+        if (cacheKey) {
+          cache.set(cacheKey, items);
+          cache.set(`${cacheKey}_lastUpdated`, Date.now().toString());
+        }
+        console.log(`fetch ${cacheKey} data success`);
+        setData(JSON.parse(items));
+      } catch (error: unknown) {
+        setError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    };
 
-      if (cacheKey) {
-        cache.set(cacheKey, items);
-        return setData(JSON.parse(cache.get(cacheKey) as string));
+    if (cacheKey && cache.has(cacheKey)) {
+      const lastUpdatedString = cache.get(`${cacheKey}_lastUpdated`) as string;
```

Comment:
I don't think this will work correctly the first time it runs on an existing install, where the cache contains a `cacheKey` element but `${cacheKey}_lastUpdated` doesn't exist.

We could guard against that by checking for both keys in the line above:

```ts
if (cacheKey && cache.has(cacheKey) && cache.has(`${cacheKey}_lastUpdated`)) {
```

Reply:
It does have some problems, but it will works correctly.
when `${cacheKey}_lastUpdated` doesn't exist, it will be `Nan`in`lastUpdated` and `timeSinceLastUpdate`.
it will always get `false` in `if (timeSinceLastUpdate < cacheTimeout)` .

but I still fix it